SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/DynamicSubobjectsTest/DynamicSubobjectsTest.cpp

```diff
@@ -0,0 +1,243 @@
+			[this](float DeltaTime) {
+				ASpatialFunctionalTestFlowController* FlowController = GetLocalFlowController();
+				APlayerController* PlayerController = Cast<APlayerController>(FlowController->GetOwner());
+				ATestMovementCharacter* PlayerCharacter = Cast<ATestMovementCharacter>(PlayerController->GetPawn());
+				if (IsValid(PlayerController))
+				{
+					if (PlayerCharacter == PlayerController->AcknowledgedPawn)
+					{
+						FinishStep();
+					}
+				}
+			},
+			StepTimeLimit);
+	}
+
+	for (int i = 0; i < DynamicComponentsPerClass + 1; ++i)
+	{
+		{ // Step 3 - Server moves the TestMovementCharacter of Client 1 to a remote location, so that it does not see the
+		  // AReplicatedGASTestActor.
+			AddStep(TEXT("DynamicSubobjectsTestServerMoveClient1"), FWorkerDefinition::Server(1), nullptr, [this, i]() {
+				if (ClientOneSpawnedPawn->SetActorLocation(CharacterRemoteLocation))
+				{
+					if (ClientOneSpawnedPawn->GetActorLocation().Equals(CharacterRemoteLocation, 1.0f))
+					{
+						FinishStep();
+					}
+				}
+			});
+		}
+
+		{ // Step 4 - Client 1 makes sure that the movement was correctly replicated
+			AddStep(
+				TEXT("DynamicSubobjectsTestClientCheckFirstMovement"), FWorkerDefinition::Client(1), nullptr, nullptr,
+				[this](float DeltaTime) {
+					ASpatialFunctionalTestFlowController* FlowController = GetLocalFlowController();
+					APlayerController* PlayerController = Cast<APlayerController>(FlowController->GetOwner());
+					ATestMovementCharacter* PlayerCharacter = Cast<ATestMovementCharacter>(PlayerController->GetPawn());
+
+					if (IsValid(PlayerCharacter))
+					{
+						if (PlayerCharacter->GetActorLocation().Equals(CharacterRemoteLocation, 50.0f))
+						{
+							FinishStep();
+						}
+					}
+				},
+				StepTimeLimit);
+		}
+
+		{ // Step 5 - Server increases AReplicatedGASTestActor's TestIntProperty to enable checking if the client is out of interest later.
+			AddStep(TEXT("DynamicSubobjectsTestServerIncreasesIntValue"), FWorkerDefinition::Server(1), nullptr, [this, i]() {
+				TArray<AActor*> FoundActors;
+				UGameplayStatics::GetAllActorsOfClass(GetWorld(), AReplicatedGASTestActor::StaticClass(), FoundActors);
+				if (FoundActors.Num() == 1)
+				{
+					TestActor = Cast<AReplicatedGASTestActor>(FoundActors[0]);
+					TestActor->TestIntProperty = i + 1;
+				}
+				if (TestActor->TestIntProperty == i + 1)
+				{
+					FinishStep();
+				}
+			});
+		}
+
+		{ // Step 6 - Client 1 checks it can no longer see the AReplicatedDynamicSubobjectsTestActor
+			AddStep(
+				TEXT("DynamicSubobjectsTestClientCheckIntValueIncreased"), FWorkerDefinition::Client(1), nullptr, nullptr,
+				[this, i](float DeltaTime) {
+					TArray<AActor*> FoundActors;
+					UGameplayStatics::GetAllActorsOfClass(GetWorld(), AReplicatedGASTestActor::StaticClass(), FoundActors);
+					if (FoundActors.Num() == 1)
+					{
+						TestActor = Cast<AReplicatedGASTestActor>(FoundActors[0]);
+						if (TestActor->TestIntProperty < i + 1)
```

Comment:
There's a conceptual problem with this check, because in theory it can sync that TestIntProperty slower than it gets to this step.

So when doing this kind of stuff I think you need to focus on doing them in another way (which is slower and should be avoided but some times we can't), ie:
Have some timer in the class and then do a tick:
RequireNotEqual(TEXT("Check TestIntProperty didn't get replicated"), TestActor->TestIntProperty, i+1);
StepTimer += DeltaTime;
if( StepTimer >= 0.5f )
{
     FinishStep();
     StepTimer = 0.0f; // reset for the next time
}